### ISSUE TITLE
Plugin refactor RFC

### ISSUE THREAD

**Issue by juniorxsound** [juniorxsound/Depthkit.js#36]
It's been a moment since this plugin was last maintained and quite a lot has changed so wanted to collect some thoughts on potential future improvements.
- [ ] New build system (perhaps Vite) to produce an ES module and externalizing three.js
- [ ] Typescript
- [ ] Build plugin in Github action on commit to avoid having to checkin built plugin file (i.e `./build/depthkit.js`)
- [ ] Deploy npm package via Github action on tag
- [ ] Add support for new Depthkit metadata structure (i.e #24)
- [ ] Better examples for WebXR API usage

**Comment by remmel** [comment]
I can migrate the code to es6 / esm (A build system won't be needed in that case). However I can also add a webpack build system to create an UMD build (w/o THREE) for thoses who don't want to use esm from browser (the only usage I see is, when using aframe lib direcly without aframe-dephkit component ; note that es6 as es-module-shims make esm available ~everywhere)

**Comment by ROBYER1** [comment]
npm package + Typescript + Vite plugin system as an ES module would be essential for my work projects using [Needle Tools](https://needle.tools/) and other things using Vite. 

Support for newer depthkit metadata structure is also essential for use of up to date versions. Thanks for your work on this!

**Comment by remmel** [comment]
@ROBYER1 Why do you need depthkit to use typescript and vite? Your project should not be dependent on the build system of the library it uses. For example you are using THREE which is built with esm and rollup

**Comment by ROBYER1** [comment]
> @ROBYER1 Why do you need depthkit to use typescript and vite? Your project should not be dependent on the build system of the library it uses. For example you are using THREE which is built with esm and rollup

Makes sense to me, I'm still figuring out a lot of Vite but thanks for the explanation!
